### PR TITLE
[WIP] ROS2 Porting: scenario_intersection

### PR DESCRIPTION
**Pull request by jilaada**
## Summary

Relatively straightforward port of the scenario_intersection package. Some changes are not complete yet due to unmet dependencies.

Notable Changes:
* rclcpp::Logger object needed to be propagated to lower level classes and methods in order for the ROS logging calls to be retained
* ros::Time is passed in as a method but not used - this was removed
* rosbag package was listed as a dependency in the package.xml - this was not seen anywhere and therefore removed
* Rearranged headers so they were only where they were needed

### Note to reviewer
Waiting on **scenario_api**

## Testing

This package should compile and at the moment with 
```
colcon build --packages-up-to scenario_intersection
```

### CODE REVIEW

File: entity/scenario_intersection/include/scenario_intersection/intersection.h [hunk 1]
```diff
@@ -66,40 +68,40 @@ class Intersection
...
-                    target_, boost::lexical_cast<std::string>(each), false);
-              });
-        }
-      }
-
-      bool operator()(ScenarioAPI& simulator) const
-      {
-        return changeColor(simulator) and changeArrow(simulator);
-      }
+      // bool changeColor(ScenarioAPI& simulator) const
```
Comment (esteve) [inline entity/scenario_intersection/include/scenario_intersection/intersection.h:71]: Why is this code commented?

**Comment by esteve** [comment]
@jilaada is this ready for review? I see that @mitsudome-r and I are assigned as reviewers.

File: entity/scenario_intersection/include/scenario_intersection/intersection.h (hunk 1 above)
Comment (jilaada) [reply entity/scenario_intersection/include/scenario_intersection/intersection.h:71]: It has been commented out as the `ScenarioAPI` class is currently not available. Once those changes are merged this branch can be rebased and references to `ScenarioAPI` can be add back.

**Comment by jilaada** [comment]
> @jilaada is this ready for review? I see that @mitsudome-r and I are assigned as reviewers.

It is set to draft. I've ported as much as I can while waiting for dependencies to be merged. Feel free to review the sections of code which have already been ported - notably the changes I listed in PR description. I figured I would create the PR for visibility as this package is one that many other depend on.

**Comment by jilaada** [comment]
Closing as this is now replaced with #26

**Pull request closed without merging**